lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -148,45 +147,62 @@ const AbstractDialog = forwardRef<HTMLDialogElement, Props>((props, ref) => {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       const dialogEl = dialogRef.current!;
 
-      // When tab key is pressed
-      // cycles focus on first or last focusable item (if exists)
-      // Do the same if focus is not IN the dialog to ensure focus is trapped within.
-      if (event.key === 'Tab') {
-        const firstFocusableElement = getFirstVisibleFocusableElement(dialogEl);
-        const lastFocusableElement = getLastVisibleFocusableElement(dialogEl);
-        const activeFocusIsNullOrDialog
-          = [document.body, dialogEl, null].includes(document.activeElement as HTMLElement);
-        const activeFocusIsInDialog = !activeFocusIsNullOrDialog && dialogEl.contains(document.activeElement);
-
-        if (!activeFocusIsInDialog) {
-          event.preventDefault();
-        }
-
-        if (event.shiftKey) {
-          if (
-            (!activeFocusIsInDialog || document.activeElement === firstFocusableElement)
-            && lastFocusableElement
-          ) {
-            lastFocusableElement.focus();
-            event.preventDefault();
+      // Only manage tab cycling when the focus is within the dialog
+      // and not nested inside another dialog
+      // so that there is no keynav conflict when another modal is open.
+      if (
+        document.activeElement
+        && document.activeElement.closest(DIALOG_MODAL_SELECTOR) === dialogEl
+      ) {
+        // Cycle focus on the first or last focusable item (if exists).
+        // Do the same if focus is NOT in the dialog to ensure that the focus is trapped within.
+        if (event.key === 'Tab') {
+          const firstFocusableElement = getFirstVisibleFocusableElement(dialogEl);
+          const lastFocusableElement = getLastVisibleFocusableElement(dialogEl);
+
+          if (event.shiftKey) {
+            if (document.activeElement === firstFocusableElement) {
+              lastFocusableElement.focus();
+              event.preventDefault();
+            }
           }
-        }
-        else {
-          if (
-            (!activeFocusIsInDialog || document.activeElement === lastFocusableElement)
-            && firstFocusableElement
-          ) {
+          else if (document.activeElement === lastFocusableElement) {
             firstFocusableElement.focus();
             event.preventDefault();
           }
         }
       }
     };
 
+    // Prevent leak when focus goes out of the dialog modal.
+    const handleFocusOut = (event: FocusEvent) => {
+      const dialogEl = dialogRef.current;
+      if (dialogEl) {
+        const departingFocusIsInsideDialog = !!(
+          event.target
+          && (event.target as HTMLElement).closest(DIALOG_MODAL_SELECTOR) === dialogEl
+        );
+        const receivingFocusIsInsideDialog
+          = !!(
+            event.relatedTarget
+            && (event.relatedTarget as HTMLElement).closest(DIALOG_MODAL_SELECTOR)
```

Comment:
Isn't this supposed to be 
```suggestion
            && (event.relatedTarget as HTMLElement).closest(DIALOG_MODAL_SELECTOR) === dialogEl
```

Reply:
> Isn't this supposed to be

Oh this was intentional change to fix the focusing issue in modals that are siblings.

Reply:
> Oh this was intentional change to fix the focusing issue in modals that are siblings.

Could you help me understand what the issue was and why this helped?  I'd think that regardless of any sibling modals, we'd still want to trap focus within this modal, and would want `focusIsLeavingDialog` to be true even if the `relatedTarget` is in a sibling modal.

Reply:
> > Oh this was intentional change to fix the focusing issue in modals that are siblings.
> 
> Could you help me understand what the issue was and why this helped? I'd think that regardless of any sibling modals, we'd still want to trap focus within this modal, and would want `focusIsLeavingDialog` to be true even if the `relatedTarget` is in a sibling modal.

Sorry I forgot to add a comment, I'll do that now:

The problem was that when you open a nested modal, this is triggered because the relatedTarget parent dialog is not the wrapping modal's dialog and `getFirstVisibleFocusableElement(dialogEl).focus();` is triggered. 

For some reason, in Chrome, this forces the focus back on the parent dialog instead of the nested dialog even though the getFirstVisibleFocusableElement on the nested dialog was triggered after. 🤔 

So this workaround only triggers when the receiving focus is not inside any modal dialog, which happens for when the focused element is hidden via CSS.

Reply:
> Sorry I forgot to add a comment, I'll do that now:
>
> The problem was that when you open a nested modal, this is triggered because the relatedTarget parent dialog is not the wrapping modal's dialog and getFirstVisibleFocusableElement(dialogEl).focus(); is triggered.
>
> For some reason, in Chrome, (This works fine in Firefox strangely), this forces the focus back on the parent dialog instead of the nested dialog even though the getFirstVisibleFocusableElement on the nested dialog was triggered after. 
>
> So this workaround only triggers when the receiving focus is not inside any modal dialog, which happens for when the focused element is hidden via CSS.

Let me check my understanding: so when one modal is nested inside another, and a `focusout` event occurs within the inner one, I imagine it triggers this handler on both modals.  And the handler for the outer modal thinks focus is leaving, since the nearest modal is the child and not itself, and so it steals the focus.  Is that right?

I wonder if a better solution would be to use `stopPropagation` in the inner modal's handler of for this event.

Reply:
> > Sorry I forgot to add a comment, I'll do that now:
> > The problem was that when you open a nested modal, this is triggered because the relatedTarget parent dialog is not the wrapping modal's dialog and getFirstVisibleFocusableElement(dialogEl).focus(); is triggered.
> > For some reason, in Chrome, (This works fine in Firefox strangely), this forces the focus back on the parent dialog instead of the nested dialog even though the getFirstVisibleFocusableElement on the nested dialog was triggered after.
> > So this workaround only triggers when the receiving focus is not inside any modal dialog, which happens for when the focused element is hidden via CSS.
> 
> Let me check my understanding: so when one modal is nested inside another, and a `focusout` event occurs within the inner one, I imagine it triggers this handler on both modals. And the handler for the outer modal thinks focus is leaving, since the nearest modal is the child and not itself, and so it steals the focus. Is that right?
> 
> I wonder if a better solution would be to use `stopPropagation` in the inner modal's handler of for this event.

Oh it's not a problem when the modals are nested inside, the problem occur when the modals are siblings.
So when the second modal opens, focusout event occurs on the first modal and calls the focus back,
It looks like looking at the dev tool that it calls it twice, both from the first modal handler, then the on mount focus gets called from the second modal but that didn't override the focusout handlers for some reason (and only on Chrome)

Going to dig into this more 🤔

Reply:
After investigating this more, it looks like that is the case:
When you open a second modal that is sibling from the other modal,
focusout is triggered and due to a race condition, steals the focus from the first focus on mount (in Chrome).

The fix here works because now the focus out handler is only triggered when the focus goes out of _any_ dialog modal. (This is a scenario that happens when the focus leaves an element when it is hidden via CSS) which was the purpose for that handler anyways.

I cleaned up the code and added a comment, let me know what your thoughts on this 👍

Reply:
> After investigating this more, it looks like that is the case:
When you open a second modal that is sibling from the other modal,
focusout is triggered and due to a race condition, steals the focus from the first focus on mount (in Chrome).
>
> The fix here works because now the focus out handler is only triggered when the focus goes out of any dialog modal. (This is a scenario that happens when the focus leaves an element when it is hidden via CSS) which was the purpose for that handler anyways.
>
> I cleaned up the code and added a comment, let me know what your thoughts on this +1

OK I think I understand now.  How about just changing the variable name to `receivingFocusIsInsideAnyDialogModal`.  "Any" is more accurate than "Another" here if I understand correctly.